mirrord/layer/src/common.rs

```diff
@@ -11,29 +12,77 @@ use crate::{
     HOOK_SENDER,
 };
 
+/// Type alias for a queue of responses from the agent, where these responses are [`RemoteResult`]s.
+///
+/// ## Usage
+///
+/// Most mirrord features that interact with the agent use this type to keep a queue of the
+/// operations, keeping them in order, and allowing us to match a hook request to a response. It's
+/// done in this way due to us not having identifiers for the hook requests.
+///
+/// - The usual flow is:
```

Comment:
Rewrite this following our chat on Discord

Reply:
Copied the discord message, should I remove the "The usual flow" part?

Reply:
Oops my mistake, disregard